src/codeu/chat/util/Tokenizer.java

```diff
@@ -0,0 +1,65 @@
+package codeu.chat.util;
+import java.io.*;
+public final class Tokenizer {
+	private StringBuilder token;
+	private String source;
+	private int at;
+
+	public Tokenizer(String source) {
+		this.source = source;
+	}
+
+	public String next() throws IOException {
+		while (remaining() > 0 && Character.isWhitespace(peek())) {
+			read();
+		}
+		if (remaining() <= 0) {
+			return null;
+		} else if (peek() == '"') {
+			//read a token that is surrounded by quotes
+			return readWithQuotes();
+		} else {
+			//read a token that is not surrounded by quotes
+			return readWithNoQuotes();
+		}
+	}
+
+	private int remaining() {
+		return source.length() - at;
+	}
+
+	private char peek() throws IOException {
+		if (at < source.length()) {
+			return source.charAt(at);
+		} else {
+			throw new IOException("too short");
+		}
+	}
+
+	private char read() throws IOException {
+		final char c = peek();
+		at += 1;
+		return c;
+	}
+
+	private String readWithNoQuotes() throws IOException {
+		token.setLength(0); //clear the token
```

Comment:
You will hit a `NullPointerException` here because `token` is never initialized anywhere.